stock_landed_costs_purchase_auto/models/purchase_order.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    landed_cost_ids = fields.One2many(
+        comodel_name="stock.landed.cost",
+        inverse_name="purchase_id",
+        string="Landed Costs",
+    )
+    landed_cost_number = fields.Integer(compute="_compute_landed_cost_number")
+
+    def _compute_landed_cost_number(self):
+        for item in self:
+            item.landed_cost_number = len(item.landed_cost_ids)
+
+    def _prepare_landed_cost_values(self):
+        return {"purchase_id": self.id, "picking_ids": self.picking_ids}
```

Comment:
Don't add all existing pickings, but only the one that is being created, so you need to pass such picking as argument, and you have to indicate with the proper command the ID.